src/libopensc/card-dnie.c

```diff
@@ -304,10 +296,17 @@ int dnie_ask_user_consent(struct sc_card * card, const char *title, const char *
 		close(srv_send[1]);
 		close(srv_recv[0]);
 		close(srv_recv[1]);
-		/* call exec() with proper user_consent_app from configuration */
-		/* if ok should never return */
-		execlp(GET_DNIE_UI_CTX(card).user_consent_app, GET_DNIE_UI_CTX(card).user_consent_app, (char *)NULL);
-		sc_log(card->ctx, "execlp() error");
+		/* check that user_consent_app exists. TODO: check if executable */
+		res = stat(GET_DNIE_UI_CTX(card).user_consent_app, &st_file);
+		if (res != 0) {
+			sc_log(card->ctx, "Invalid pinentry application: %s\n",
+					GET_DNIE_UI_CTX(card).user_consent_app);
+		} else {
+			/* call exec() with proper user_consent_app from configuration */
+			/* if ok should never return */
+			execlp(GET_DNIE_UI_CTX(card).user_consent_app, GET_DNIE_UI_CTX(card).user_consent_app, (char *)NULL);
+			sc_log(card->ctx, "execlp() error");
+		}
```

Comment:
nit: The expression `GET_DNIE_UI_CTX(card).user_consent_app` is used in the above code already 4 times. I think it should already deserve temporary variable to improve readability.